src/helpers.js

```diff
@@ -61,6 +65,64 @@ function readFileString (file) {
     })
   })
 }
+/**
+ * Read a cached cert when available or get a new one.
+ * @private
+ * @param {string} env IMS environment type
+ * @param {string} certName certificate that can be used to verify the signature
+ * @returns {Buffer | Promise<Buffer>} cert
+ */
+async function _readOrGetCert (env, certName) {
+  try {
+    return fs.readFileSync(path.join(CONFIG_PATH, certName))
+  } catch (e) {
+    if (env && certName) {
+      const URL = `https://static.adobelogin.com/keys/${env}/${certName}`
```

Comment:
let's use https://github.com/adobe/aio-lib-core-networking with exponential backoff and use async/await.